### DISCUSSION TITLE
How can you scroll parent container when item inside is extended towards either of the sides

### DISCUSSION THREAD

**Discussion by Ahmed-Ateeq** [samuelarbibe/dnd-timeline#48] [Q&A] [answered]
My container for timeline rows can be greater in width than its wrapper, meaning that it can be scrollable. However, when items are resized towards either right or the left side, then I want to smoothly scroll my container so the item keeps resizing if kept in lets say the last 60pixels of the container.

Any ideas on how I can achieve this? The attached video shows the behavior that I want. 

https://github.com/user-attachments/assets/532f1252-effd-48c6-b84d-42f4351ea512

**Comment by samuelarbibe** [comment]
Hey Ahmed!
If I understand correctly, you wish to "scroll" horizontally when dragging approaches or overflows one of the timeline's sides.  
  
I suggest you look at my "virtual" example, where I implemented the same behavior for vertical scrolling.  
  
https://github.com/samuelarbibe/dnd-timeline/blob/main/examples/virtual/src/useAutoscroll.ts
  
The only difference is, that you don't horizontally scroll, you simulate it by modifying the range property to "move" left or right.  
  
Good luck, and don't hesitate to ask for further help.

**Reply by Ahmed-Ateeq** [reply]
Ah, I get the idea.. thanks!

In my case, I want to have the scrolling effect on resizing an item. The issue Im facing now is that I'm unable to sync the item drag handle along with my cursor and the scroll effect. 

My basic structure for timeline is: [sandbox](https://codesandbox.io/p/sandbox/sweet-sammet-97jdxq?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clzjnb1og00063b6lxzsshubw%2522%252C%2522sizes%2522%253A%255B100%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clzjnb1og00023b6l8ajwyv99%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clzjnb1og00033b6liqzsub0f%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clzjnb1og00053b6l9n2xa5l4%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clzjnb1og00023b6l8ajwyv99%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clzjnb1of00013b6lhz8e0poy%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clzjp4ryh00023b6lb1rtpkkm%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A27%252C%2522startColumn%2522%253A14%252C%2522endLineNumber%2522%253A27%252C%2522endColumn%2522%253A14%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fcomponents%252Ftimeline%252Ftimeline.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clzjnb1og00023b6l8ajwyv99%2522%252C%2522activeTabId%2522%253A%2522clzjp4ryh00023b6lb1rtpkkm%2522%257D%252C%2522clzjnb1og00053b6l9n2xa5l4%2522%253A%257B%2522id%2522%253A%2522clzjnb1og00053b6l9n2xa5l4%2522%252C%2522activeTabId%2522%253A%2522clzjo4v7m00153b6lmb9y856t%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clzjnb1og00043b6lhf1b5lwr%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clzjo4v7m00153b6lmb9y856t%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clzjnb1og00033b6liqzsub0f%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clzjnb1og00033b6liqzsub0f%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

I have a fixed set of width defined per second. Im not entirely sure on how I can increase range and consistently keep resizing my items along with that.

**Reply by samuelarbibe** [reply, marked as answer]
I would suggest you to separate the scrolling logic, and activate it only while resizing.  
  
If you want to know when you are currently dragging, you can use the `onResizeStart` and `onResizeEnd` callbacks to manage a custom `isDragging` state, and activate the `useAutoscroll` mechanism, which already handles mouse positioning, only when `isDragging` is true.
